recipes/resiprocate/cmake/conanfile.py

```diff
@@ -0,0 +1,151 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import is_apple_os
+from conan.tools.build import check_min_cppstd, cross_building
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rm, rmdir
+from conan.tools.gnu import PkgConfigDeps
+from conan.tools.microsoft import is_msvc
+
+required_conan_version = ">=2.0.9"
+
+
+class ResiprocateConan(ConanFile):
+    name = "resiprocate"
+    description = (
+        "The project is dedicated to maintaining a complete, correct, "
+        "and commercially usable implementation of SIP and a few related protocols."
+    )
+    license = "VSL-1.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/resiprocate/resiprocate/wiki/"
+    topics = ("sip", "voip", "communication", "signaling")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cares": [True, False],
+        "with_ssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_cares": True,
+        "with_ssl": True,
+    }
+    implements = ["auto_shared_fpic"]
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.with_cares:
+            self.requires("c-ares/[>=1.27 <2]")
+        if self.options.with_ssl:
+            self.requires("openssl/[>=1.1 <4]")
+
+    def validate(self):
+        check_min_cppstd(self, 11)
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on msvc, it does not export required symbols.")
```

Comment:
@AbrilRBS and I were trying to compile it as shared and we saw those export symbol errors, that's why we updated the message. @gjasny Is this the real problem? Are there any open issues related to this?